src/ECharts/Events.purs

```diff
@@ -0,0 +1,68 @@
+module ECharts.Events (
+  EventType(..),
+  EventParam(),
+  Sub(),
+  listen
+  ) where
+
+import Data.Function
+import Control.Monad.Eff
+import Data.Argonaut.Core
+
+import ECharts.Chart
+import ECharts.Effects
+
+data EventType = Refresh | Restore | Resize | Click | DoubleClick | Hover
+               | DataChanged | DataZoom | DataRange | DataRangeHoverLink
+               | LegendSelected | LegendHoverLink | MapSelected | PieSelected
+               | DataViewChanged | MapRoam | MagicTypeChanged
+
+type EventParam = Json
+
+newtype Sub = Sub (forall eff. Eff (unlisten :: Unlisten|eff) Unit)
+
+instance eventTypeShow :: Show EventType where
```

Comment:
Again, `Show` should render things as Purescript code. Ordinary functions can be used for other types of renderings (e.g. `eventType :: EventType -> String`).